common/lib/transport/connectionmanager.js

```diff
@@ -1,9 +1,8 @@
 var ConnectionManager = (function() {
-	var readCookie = (typeof(Cookie) !== 'undefined' && Cookie.read);
-	var createCookie = (typeof(Cookie) !== 'undefined' && Cookie.create);
-	var eraseCookie = (typeof(Cookie) !== 'undefined' && Cookie.erase);
-	var connectionKeyCookie = 'ably-connection-key';
-	var connectionSerialCookie = 'ably-connection-serial';
+	var getFromSession    = (typeof(SessionStorage) !== 'undefined' && SessionStorage.get);
```

Comment:
I find this approach a little odd.  Why are we creating a `SessionStorage` object that sometimes has an interface with `get`, `set` and `remove`, and other times without the method.  I would prefer we either:

* Have a method that on SessionStorage such as `isSupported()` but the interface remains the same.
* My preference: If `sessionStorage` is not supported, we should have `noop` methods in their place that simply don't do anything.